src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -434,7 +434,12 @@ private AssemblyResult createGraph(final Iterable<GATKRead> reads,
             return null;
         }
 
-        return getAssemblyResult(refHaplotype, kmerSize, rtgraph, aligner);
+        final AssemblyResult result = getAssemblyResult(refHaplotype, kmerSize, rtgraph, aligner);
+        // check whether recovering dangling ends created cycles
+        if (recoverAllDanglingBranches && rtgraph.hasCycles()) {
```

Comment:
It doesn't actually matter if `recoverAllDanglingBranches` right? You want to return null if there are cycles regardless?

Reply:
You're right but I'm being conservative about runtime here.  This is all going to be overhauled soon anyway.

Reply:
And don't worry about the code coverage.  It's effectively a random number generator.

Reply:
Sounds good, merge away!